python/ml_croissant/ml_croissant/_src/structure_graph/nodes/source.py

```diff
@@ -28,6 +31,32 @@ def parse_reference(issues: Issues, source_data: str) -> tuple[str, ...]:
     return groups
 
 
+class FileProperty(enum.Enum):
+    """Lists the intrinsic properties of a file that are accessible from Croissant."""
+
+    content = enum.auto()
+    filename = enum.auto()
+    filepath = enum.auto()
+    fullpath = enum.auto()
+
+
+def is_file_property(file_property: str):
```

Comment:
Might be nice to add a test for this if time allows?

Reply:
Thanks! Just did it in another PR: https://github.com/mlcommons/croissant/pull/150/commits/3f1b79824fa42c289e38a44fc192905d87101c01